### PR TITLE
Adding basic API documentation for List, String, Number and Boolean

### LINKED ISSUE #25: List API documentation

### PR DESCRIPTION
**Pull request by taras**
Closes #25, #24, #26, #27

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -166,40 +170,248 @@ Remove a key (and subsequent value) from this object. For example, to delete the
...
+
+Add property with a given name and value to the object. It will update the property with the given value if it already exists.
+
+```handlebars
+<button onclick={{action car.put "color" "blue"}}>
+  Add color
+</button>
+```
+
+#### `set(obj)`
```
Comment (cowboyd) [inline README.md]: In order for the link to work, I think this needs to match the link above (minus any special characters). So in this case the link is `[set(object)](#setvalue)` and the heading is `set(obj)`

A good match might be `[set(object)](#setobject)`

File: README.md [hunk 2]
```diff
@@ -166,40 +170,248 @@ Remove a key (and subsequent value) from this object. For example, to delete the
...
+
+```handlebars
+<button onclick={{action car.set (hash make="Toyota" model="Supra" year="1982")}}>
+  Update car
+</button>
+```
 
 ### `List`
 
+List microstate represents an indexed array of values.
```
Comment (cowboyd) [inline README.md]: I don't think we have an API currently for accessing values by index do we? I'd say the main thing about lists is that they're an ordered collection.

File: README.md [hunk 3]
```diff
@@ -166,40 +170,248 @@ Remove a key (and subsequent value) from this object. For example, to delete the
...
+{{let numbers=(List (array 1 2 3))}}
+
+{{each numbers as |item|}}
+  {{item}}
+{{/each}}
+```
+
 #### `concat(list)`
+
+Merges given array with existing list.
```
Comment (cowboyd) [inline README.md]: Is merge really the right word here? Seems like what's happening is that we're making a new list with all of the items from `list` added to the end of the current list.

File: README.md (hunk 2 above)
Comment (taras) [inline README.md]: Yep, updated.

File: README.md (hunk 3 above)
Comment (taras) [inline README.md]: Yeah, `making a new list with all of the items from list added to the end of the current list` is a better description. I updated to use that description for all list transitions.

**Comment by taras** [comment]
@cowboyd should be good to go.

**Pull request merged**